DigitalLearningSolutions.Data/Services/GroupsService.cs

```diff
@@ -400,6 +427,52 @@ int centreId
             transaction.Complete();
         }
 
+        public void GenerateGroupsFromRegistrationField(GroupGenerationDetails groupDetails)
+        {
+            var isJobGroup = groupDetails.LinkedToField == 7;
+
+            (List<(int id, string name)> newGroupNames, string groupNamePrefix) = isJobGroup
+                ? GetJobGroupsAndPrefix()
+                : GetCustomPromptsAndPrefix(groupDetails.CentreId, groupDetails.LinkedToField);
+
+            var groupsAtCentre = GetGroupsForCentre(groupDetails.CentreId).Select(g => g.GroupLabel).ToList();
+
+            using var transaction = new TransactionScope();
+            foreach (var (id, newGroupName) in newGroupNames)
+            {
+                var groupName = groupDetails.PrefixGroupName ? $"{groupNamePrefix} - {newGroupName}" : newGroupName;
+
+                if (groupDetails.SkipDuplicateNames && groupsAtCentre.Contains(groupName))
+                {
+                    return;

```

Comment:
Right, if one group is skipped because it has a duplicate name, we still want to create the other groups. I've changed 'return' to 'continue', thanks!

Not sure about case sensitivity, I'll default to @SteveJacksonSoft to answer this

Reply:
Yes this is only for multiple choice prompts, and it is case-sensitive already so I'll keep it that way.

Added a check to the POST to make sure the field is not a free text one.